### PR TITLE
Feat - Set up @Output for click in holdings table component #3835

### PR DESCRIPTION
**Pull request by Prakhar29Sharma**
- Replaced direct routing in GfHoldingsTableComponent with an @Output() event emitter.
- Emitting 'holdingClicked' event to parent component when a holding is clicked.
- Added event listener in HomeHoldingsComponent to handle holding click and perform actions (e.g., navigation).
- Improved separation of concerns between child and parent components.

### CODE REVIEW

File: apps/client/src/app/components/home-holdings/home-holdings.component.ts [hunk 1]
```diff
@@ -119,6 +119,14 @@ export class HomeHoldingsComponent implements OnDestroy, OnInit {
     }
   }
 
+  public onHoldingClicked({ dataSource, symbol }: AssetProfileIdentifier) {
```
Comment (dtslvr) [inline apps/client/src/app/components/home-holdings/home-holdings.component.ts:122]: Please move the method up that the methods are sorted alphabetically.

File: apps/client/src/app/components/home-holdings/home-holdings.component.ts [hunk 2]
```diff
@@ -119,6 +119,14 @@ export class HomeHoldingsComponent implements OnDestroy, OnInit {
...
+    if (dataSource && symbol) {
+      this.router.navigate([], {
+        queryParams: { dataSource: dataSource, symbol: symbol, holdingDetailDialog: true }
```
Comment (dtslvr) [inline apps/client/src/app/components/home-holdings/home-holdings.component.ts:125]: Please use the short form:

```
queryParams: { dataSource, symbol, holdingDetailDialog: true }
```

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thank you for the contribution, @Prakhar29Sharma!

I have added a few comments.

Please run `npm run nx format` to automatically format your code so it meets the project's formatting standards and passes the pipeline checks 🙂

File: apps/client/src/app/components/home-holdings/home-holdings.component.ts (hunk 1 above)
Comment (Prakhar29Sharma) [reply apps/client/src/app/components/home-holdings/home-holdings.component.ts:122]: > Please move the method up that the methods are sorted alphabetically.

Hey @dtslvr, Thanks for the feedback. Can you please clarify where exactly it should be present, Should I place the method below onChangeHoldingType() method?

Comment (dtslvr) [reply apps/client/src/app/components/home-holdings/home-holdings.component.ts:122]: Yes, exactly 👍🏻

File: package-lock.json
Comment (dtslvr) [inline package-lock.json:1]: Please revert these changes

**Review by dtslvr — CHANGES_REQUESTED** [review]

**Review by dtslvr — APPROVED** [review]
Thanks again for your contribution, @Prakhar29Sharma! 👍🏻
I have made some final adjustments. Ready to merge now.
